src/cinder/gl/GlslProg.cpp

```diff
@@ -697,10 +718,20 @@ void GlslProg::loadShader( const string &shaderSource, const fs::path &shaderPat
 			}
 			log += "\n" + ss.str();
 		}
-#endif		
+#endif
+		// Since the GlslProg destructor will not be called after we throw, we must delete all
+		// owned GL objects here to avoid leaking. Any other attached shaders will be cleaned up
+		// when the program is deleted
+		glDeleteShader( handle );
+		glDeleteProgram( mHandle );
```

Comment:
Yeah it's a bit of a shame, since this bails on the whole program and wouldn't allow someone catching the exception to do anything with it after that.. like say attempting to recompile a single shader. But that's not possible anyway with the current GlslProg implementation, and I don't see another way around it, unless we move compilation and linking to an init function outside the constructor.